flake.nix

```diff
@@ -8,183 +8,166 @@
-        helix = makeOverridableHelix self.packages.${system}.helix-unwrapped {};
-        default = self.packages.${system}.helix;
-      };
+        strictDeps = true;
 
-      checks = {
-        # Build the crate itself
-        inherit (self.packages.${system}) helix;
+        # Sets the Helix runtimedir to the grammars
+        env.HELIX_DEFAULT_RUNTIME = "${runtimeDir}";
 
-        clippy = craneLibMSRV.cargoClippy (commonArgs
-          // {
-            inherit cargoArtifacts;
-            cargoClippyExtraArgs = "--all-targets -- --deny warnings";
-          });
+        # Get all the application stuff in the output directory.
+        postInstall = ''
+          mkdir -p $out/lib
+          installShellCompletion ${./contrib/completion}/hx.{bash,fish,zsh}
+          mkdir -p $out/share/{applications,icons/hicolor/256x256/apps}
+          cp ${./contrib/Helix.desktop} $out/share/applications
+          cp ${./contrib/helix.png} $out/share/icons/hicolor/256x256/apps
+        '';
 
-        fmt = craneLibMSRV.cargoFmt commonArgs;
+        meta.mainProgram = "hx";
+      });
+  in
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [(import rust-overlay)];
+      };
 
-        doc = craneLibMSRV.cargoDoc (commonArgs
-          // {
-            inherit cargoArtifacts;
-          });
+      # Get Helix's MSRV toolchain to build with by default.
+      msrvToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+      msrvPlatform = pkgs.makeRustPlatform {
+        cargo = msrvToolchain;
+        rustc = msrvToolchain;
+      };
+    in {
+      packages = rec {
+        # Make MSRV Helix
+        helix = pkgs.callPackage mkHelix {rustPlatform = msrvPlatform;};
 
-        test = craneLibMSRV.cargoTest (commonArgs
-          // {
-            inherit cargoArtifacts;
-          });
+        # The default Helix build. Uses the default MSRV Rust toolchain, and the
+        # default nixpkgs, which is the one in the Flake.lock of Helix.
+        #
+        # This can be overridden though to add Cargo Features, flags, and different toolchains.
+        default = helix;
       };
 
-      devShells.default = pkgs.mkShell {
-        inputsFrom = builtins.attrValues self.checks.${system};
-        nativeBuildInputs = with pkgs;
-          [lld_13 cargo-flamegraph rust-analyzer]
-          ++ (lib.optional (stdenv.isx86_64 && stdenv.isLinux) pkgs.cargo-tarpaulin)
-          ++ (lib.optional stdenv.isLinux pkgs.lldb)
-          ++ (lib.optional stdenv.isDarwin pkgs.darwin.apple_sdk.frameworks.CoreFoundation);
-        shellHook = ''
-          export HELIX_RUNTIME="$PWD/runtime"
-          export RUST_BACKTRACE="1"
-          export RUSTFLAGS="''${RUSTFLAGS:-""} ${rustFlagsEnv}"
-        '';
+      checks.helix = self.outputs.packages.${system}.helix.overrideAttrs {
+        cargoBuildType = "debug";
       };
+
+      formatter = pkgs.alejandra;
```

Comment:
I'd prefer to leave the formatter unconfigured for now. If I understand correctly, Nix is converging on a formatting style from an RFC and I'd like to follow that eventually if it becomes broadly accepted.

Reply:
Yeah there is the RFC style formatter too
https://github.com/NixOS/nixfmt

Reply:
I won't bother reformatting with the RFC style unless you really want to.